addons/payment_stripe/models/payment.py

```diff
@@ -49,14 +48,39 @@ def stripe_form_generate_values(self, tx_values):
             'payment_intent_data[description]': tx_values['reference'],
             'customer_email': tx_values.get('partner_email') or tx_values.get('billing_partner_email'),
         }
-        if tx_values.get('billing_partner_country').code and tx_values.get('billing_partner_country').code.lower() == 'nl' and \
-           tx_values.get('currency').name and tx_values.get('currency').name.lower() == 'eur':
-            # enable iDEAL for NL-based customers (€ payments only)
-            stripe_session_data['payment_method_types[1]'] = 'ideal'
+
+        self._add_available_payment_method_types(stripe_session_data, tx_values)
+
         tx_values['session_id'] = self._create_stripe_session(stripe_session_data)
 
         return tx_values
 
+    @api.model
+    def _add_available_payment_method_types(self, stripe_session_data: dict, tx_values: dict):
```

Comment:
It's useful because IDE will automatically offer relevant attributes when you're using the parameter.

Anyway, it needs more thought and I wouldn't want to make the codebase motley, so **I removed them**.

Reply:
@AdrienHorgnies Specifying the type of the param in the docstring will give you the same level of annotations in PyCharm

Reply:
@AntoineVDV Unfortunately, no: https://youtrack.jetbrains.com/issue/PY-40937 .

Reply:
There is indeed not an exact equivalence but it's well enough sufficient in 99% of the cases